airflow/contrib/hooks/sagemaker_hook.py

```diff
@@ -16,299 +16,793 @@
+            last_describe_job_call = time.time()
+
+            if secondary_training_status_changed(description, last_description):
+                self.log.info(secondary_training_status_message(description, last_description))
+                last_description = description
+
+            status = description['TrainingJobStatus']
+
+            if status not in non_terminal_states:
+                state = LogState.JOB_COMPLETE
+        return state, last_description, last_describe_job_call
+
+    def describe_tuning_job(self, name):
         """
-        :param tuning_job_name: the name of the tuning job
-        :type tuning_job_name: string
-        Return the tuning job info associated with the current job_name
+        Return the tuning job info associated with the name
+
+        :param name: the name of the tuning job
+        :type name: string
         :return: A dict contains all the tuning job info
         """
-        return self.conn\
-            .describe_hyper_parameter_tuning_job(
-                HyperParameterTuningJobName=tuning_job_name)
 
-    def describe_transform_job(self, transform_job_name):
+        return self.get_conn().describe_hyper_parameter_tuning_job(
+            HyperParameterTuningJobName=name)
+
+    def describe_model(self, name):
+        """
+        Return the SageMaker model info associated with the name
+
+        :param name: the name of the SageMaker model
+        :type name: string
+        :return: A dict contains all the model info
+        """
+
+        return self.get_conn().describe_model(ModelName=name)
+
+    def describe_transform_job(self, name):
         """
-        :param transform_job_name: the name of the transform job
-        :type transform_job_name: string
-        Return the transform job info associated with the current job_name
+        Return the transform job info associated with the name
+
+        :param name: the name of the transform job
+        :type name: string
         :return: A dict contains all the transform job info
         """
-        return self.conn\
-            .describe_transform_job(
-                TransformJobName=transform_job_name)
+
+        return self.get_conn().describe_transform_job(TransformJobName=name)
+
+    def describe_endpoint_config(self, name):
+        """
+        Return the endpoint config info associated with the name
+
+        :param name: the name of the endpoint config
+        :type name: string
+        :return: A dict contains all the endpoint config info
+        """
+
+        return self.get_conn().describe_endpoint_config(EndpointConfigName=name)
+
+    def describe_endpoint(self, name):
+        """
+        :param name: the name of the endpoint
+        :type name: string
+        :return: A dict contains all the endpoint info
+        """
+
+        return self.get_conn().describe_endpoint(EndpointName=name)
+
+    def check_status(self, job_name, non_terminal_states,
```

Comment:
All but one calls have non_terminal_states as `SageMakerHook.non_terminal_states`, so how about we:

Give it a default of `def check_status(self, job_name, ..., non_terminal_states=SageMakerHook.non_terminal_states)` (it has to go at the end to have a default) and then the one endpoint call that changes the value can pass `self.check_status(...., non_terminal_states=self.endpoint_non_terminal_states)`

Reply:
Updated.